morpheus/_lib/src/messages/multi_tensor.cpp

```diff
@@ -80,18 +105,37 @@ void MultiTensorMessage::get_slice_impl(std::shared_ptr<MultiMessage> new_messag
 {
     auto sliced_message = MRC_PTR_CAST(MultiTensorMessage, new_message);
 
-    sliced_message->offset = start;
+    // Start must be between [0, mess_count)
+    if (start < 0 || start >= this->count)
+    {
+        throw std::out_of_range("Invalid memory `start` argument");
+    }
+
+    // Stop must be between (start, mess_count]
+    if (stop <= start or stop > this->count)
+    {
+        throw std::out_of_range("Invalid memory `stop` argument");
+    }
```

Comment:
If I understand correctly this means start & stop are always absolute, and not relative to the offset?

Reply:
No start and stop are relative to the offset. But you are still bound by the relative count. Before you could call `get_slice(10, 20).get_slice(0, 20)` which is inconsistent because you are trying to get 20 rows from a 10 row slice. This just enforces that slices must be equal to or smaller than the parent.